indico/web/templates/forms/location_widget.html

```diff
@@ -0,0 +1,534 @@
+                field.closest('.i-location-field').toggleClass('disabled', isChecked);
+
+                hiddenData.inheriting = isChecked;
+                field.val(JSON.stringify(hiddenData)).trigger('change');
+
+                if (!isInitStep) {
+                    if (isChecked) {
+                        hiddenData = {{ field.data.source.location_data_for_ui | tojson }}
+                        hiddenData.inheriting = true;
+                        field.val(JSON.stringify(hiddenData)).trigger('change');
+                        initializeInputs();
+                    } else {
+                        clearVenue();
+                    }
+                }
+            };
+
+            function reorderCategories(venueName) {
+                var groupOrder = [venueName];
+                for (var i = venueNames.length - 1; i >= 0; i--) {
+                    if (!~groupOrder.indexOf(venueNames[i])) {
+                        groupOrder.push(venueNames[i]);
+                    }
+                }
+                Typeahead[roomInput.selector].options.groupOrder = groupOrder;
+            }
+
+            function updateVenueOnRoomChange(item) {
+                if (item.venue_id != hiddenData.venue_id) {
+                    venueJustChanged = true;
+                    var data = {item_name: item.group, venue_id: item.venue_id, room_id: item.id};
+                    venueInput.val(item.group).trigger('typeahead:click-on-option', data);
+                }
+            }
+
+            function clearRoom() {
+                delete hiddenData.room_id;
+                delete hiddenData.room_name;
+                roomInput.val('');
+                roomInput.next('.highlighter').removeClass('visible');
+                roomJustCleared = true;
+                postSelectionActions(roomInput, '');
+            }
+
+            function clearVenue() {
+                delete hiddenData.venue_id;
+                delete hiddenData.venue_name;
+                venueInput.val('');
+                venueInput.next('.highlighter').removeClass('visible');
+                venueJustChanged = true;
+                postSelectionActions(venueInput, '');
+                clearRoom();
+            }
+
+            function clearField(field) {
+                if (field.is(venueInput)) {
+                    clearVenue();
+                } else if (field.is(roomInput)) {
+                    clearRoom();
+                }
+            }
+
+            function updateResultsList(node, resultHtmlList) {
+                var resultsWrapper = node.siblings('.typeahead-results');
+                resultsWrapper.css('top', node.outerHeight() + 'px');
+                var justUse = node.parent().find('.just-use-option');
+                if (node.val().length > 0) {
+                    if (justUse.length === 0) {
+                        justUse = $('<div>').attr({
```

Comment:
why `attr()` instead of passing the dict with class/click as the second argument of `$()`?